lib/web_of_science/map_names.rb

```diff
@@ -88,18 +88,28 @@ def medline_name(name)
         name[:given_name] = given
         name[:first_name] = first if first.present?
         name[:middle_name] = middle if middle.present?
+        name[:name] = "#{name[:last_name]},#{name[:first_name]},#{name[:middle_name]}" # full name in the older style format used by Profiles/CAP
         name
       end
 
       # Parse the WOS names and return a Hash compatible with Csl::AuthorName
       # @return [Hash]
       def wos_name(name)
+        # look for the case where the first_name is the initials, with first and middle initials combined
+        # e.g. first_name = "RB", should be first_name = "R", middle_name = "B"
         match = name[:first_name].to_s.match(/\A([A-Z])([A-Z])/)
         if match
-          # first_name is the initials, with first and middle initials combined
           name[:first_name] = match[1]
           name[:middle_name] ||= match[2]
         end
+        # look for the case where the first name includes the middle initial optionally followed by a period
+        # e.g. first_name = "Russel B.", should be first_name = "Russel", middle_name = "B"
+        match = name[:first_name].to_s.match(/[\s][A-Z][.]?\z/)
```

Comment:
This regex/delete combo should be improved:
- regex has 3x as many braces as necessary (if indeed even one is required)
- the `[A-Z]` pattern fails to match Unicode characters   
- the `delete!` is dangerously inaccurate.  

You should use the regex to do the substitution (deletion) to make sure you don't end up with something like this:
```ruby
x = 'RUSSEL R. R.'
=> "RUSSEL R. R."
match = x.match(/[\s][A-Z][.]?\z/)
=> #<MatchData " R.">
x.delete!(match[0])
=> "USSEL"
x
=> "USSEL"
```

Reply:
To fix the above issues, I'd recommend something like:
```ruby
name[:first_name].gsub!(/\s[[:upper:]]\.?\z/) { |s| name[:middle_name] = s.strip; '' } # block returns empty, removing match
```

I confirmed that `[[:upper:]]]` matches unicode chars w/:
```ruby
'Ångström'.gsub(/[[:upper:]]/) {|s| puts s; '' }
Å
=> "ngström"
```